ghcide/src/Development/IDE/Plugin/Completions/Logic.hs

```diff
@@ -904,7 +904,7 @@ getCompletionPrefix pos@(Position l c) (VFS.VirtualFile _ _ ropetext) =
             lastMaybe = headMaybe . reverse
 
         -- grab the entire line the cursor is at
-        curLine <- headMaybe $ T.lines $ Rope.toText
+        curLine <- headMaybe $ Rope.lines
```

Comment:
This is supposed to be faster than Text.lines, so why not use it here as well..
https://hackage.haskell.org/package/text-rope-0.2/docs/Data-Text-Lines.html#v:lines

Reply:
Yes! We really need to audit all uses of `Rope.toText`, maybe even give it a hlint hint. It's generally unnecessarily expensive and does work proportional to the length of the document...